Lib/fontTools/cffLib/cff2_merge_funcs.py

```diff
@@ -0,0 +1,246 @@
+		fdArray = topDict.FDArray = FDArrayIndex()
+		fdArray.strings = None
+		fdArray.GlobalSubrs = topDict.GlobalSubrs
+		topDict.GlobalSubrs.fdArray = fdArray
+		charStrings = topDict.CharStrings
+		if charStrings.charStringsAreIndexed:
+			charStrings.charStringsIndex.fdArray = fdArray
+		else:
+			charStrings.fdArray = fdArray
+		fontDict = FontDict()
+		fontDict.setCFF2(True)
+		fdArray.append(fontDict)
+		fontDict.Private = privateDict
+		privateOpOrder = buildOrder(privateDictOperators2)
+		for entry in privateDictOperators:
+			key = entry[1]
+			if key not in privateOpOrder:
+				if key in privateDict.rawDict:
+					# print "Removing private dict", key
+					del privateDict.rawDict[key]
+				if hasattr(privateDict, key):
+					delattr(privateDict, key)
+					# print "Removing privateDict attr", key
+	else:
+		# clean up the PrivateDicts in the fdArray
+		fdArray = topDict.FDArray
+		privateOpOrder = buildOrder(privateDictOperators2)
+		for fontDict in fdArray:
+			fontDict.setCFF2(True)
+			for key in fontDict.rawDict.keys():
+				if key not in fontDict.order:
+					del fontDict.rawDict[key]
+					if hasattr(fontDict, key):
+						delattr(fontDict, key)
+
+			privateDict = fontDict.Private
+			for entry in privateDictOperators:
+				key = entry[1]
+				if key not in privateOpOrder:
+					if key in privateDict.rawDict:
+						# print "Removing private dict", key
+						del privateDict.rawDict[key]
+					if hasattr(privateDict, key):
+						delattr(privateDict, key)
+						# print "Removing privateDict attr", key
+	# Now delete up the decrecated topDict operators from CFF 1.0
+	for entry in topDictOperators:
+		key = entry[1]
+		if key not in opOrder:
+			if key in topDict.rawDict:
+				del topDict.rawDict[key]
+			if hasattr(topDict, key):
+				delattr(topDict, key)
+
+	# At this point, the Subrs and Charstrings are all still T2Charstring class
+	# easiest to fix this by compiling, then decompiling again
+	cff.major = 2
+	file = BytesIO()
+	cff.compile(file, otFont, isCFF2=True)
+	file.seek(0)
+	cff.decompile(file, otFont, isCFF2=True)
+
+
+def pointsDiffer(pointList):
+	return not (max(pointList) == min(pointList))
+
+
+def convertCFFtoCFF2(varFont):
+	# Convert base font to a single master CFF2 font.
+	cffTable = varFont['CFF ']
+	lib_convertCFFToCFF2(cffTable.cff, varFont)
+	newCFF2 = newTable("CFF2")
+	newCFF2.cff = cffTable.cff
+	varFont['CFF2'] = newCFF2
+	del varFont['CFF ']
+
+
+class MergeDictError(TypeError):
+	def __init__(self, key, value, values):
+		error_msg = ["For the Private Dict key '{}', ".format(key),
+					 "the default font value list:",
+					 "\t{}".format(value),
+					 "had a different number of values than a region font:"]
+		error_msg += ["\t{}".format(region_value) for region_value in values]
+		error_msg = os.linesep.join(error_msg)
+
+
+def conv_to_int(num):
+	if num % 1 == 0:
+		return int(num)
+	return num
+
+
+pd_blend_fields = {"BlueValues": 1, "OtherBlues": 1, "FamilyBlues": 1,
```

Comment:
I don't see any reason for using a dictionary when this can be accomplished with a tuple of strings.

Reply:
Done.